tests/functional/Calendar.php

```diff
@@ -35,31 +35,76 @@
 
 namespace tests\units;
 
+use CalendarSegment;
 use DbTestCase;
 
 /* Test for inc/calendar.class.php */
 
 class Calendar extends DbTestCase
 {
-    public function testComputeEndDate()
+    /**
+     * Data provider for the testComputeEndDate function
+     *
+     * @return iterable
+     */
+    protected function testComputeEndDateProvider(): iterable
     {
-        $calendar = new \Calendar();
-       //get default calendar
-        $this->boolean($calendar->getFromDB(getItemByTypeName('Calendar', 'Default', true)))->isTrue();
+        // Default calendar
+        $default_calendar = getItemByTypeName('Calendar', 'Default');
 
-       // ## test future dates
-        $end_date = $calendar->ComputeEndDate("2018-11-19 10:00:00", 7 * DAY_TIMESTAMP, 0, true);
-        $this->string($end_date)->isEqualTo("2018-11-28 10:00:00");
-       // end of day
-        $end_date = $calendar->ComputeEndDate("2018-11-19 10:00:00", 7 * DAY_TIMESTAMP, 0, true, true);
-        $this->string($end_date)->isEqualTo("2018-11-28 20:00:00");
-
-       // ## test past dates
-        $end_date = $calendar->ComputeEndDate("2018-11-19 10:00:00", -7 * DAY_TIMESTAMP, 0, true);
-        $this->string($end_date)->isEqualTo("2018-11-08 10:00:00");
-       // end of day
-        $end_date = $calendar->ComputeEndDate("2018-11-19 10:00:00", -7 * DAY_TIMESTAMP, 0, true, true);
-        $this->string($end_date)->isEqualTo("2018-11-08 20:00:00");
+        // 5x24h (monday to friday)
+        $five_by_twentyfour = $this->createItem(\Calendar::class, [
+            'name' => "5x24",
+        ]);
+        $parent = $five_by_twentyfour->getID();
+        $this->createItems(CalendarSegment::class, [
+            ['calendars_id' => $parent, 'day' => 1, 'begin' => '00:00:00', 'end' => '24:00:00'],
+            ['calendars_id' => $parent, 'day' => 2, 'begin' => '00:00:00', 'end' => '24:00:00'],
+            ['calendars_id' => $parent, 'day' => 3, 'begin' => '00:00:00', 'end' => '24:00:00'],
+            ['calendars_id' => $parent, 'day' => 4, 'begin' => '00:00:00', 'end' => '24:00:00'],
+            ['calendars_id' => $parent, 'day' => 5, 'begin' => '00:00:00', 'end' => '24:00:00'],
+        ]);
```

Comment:
IMHO, a test on a calendar with opening hours from 9 to 18 would be more relevant. Also, you should add test cases that use a date starting outside working hours/working days.

Reply:
This test case is no longer relevant as we are no longer modifying the `computeEndDate` method.
Maybe I'll revert the changes but I guess it doesn't hurt to include it.